src/ext/NetFx/wixext/NetFxCompiler.cs

```diff
@@ -325,5 +361,184 @@ private void ParseNativeImageElement(Intermediate intermediate, IntermediateSect
+                                    break;
+                                case "major":
+                                    rollForward = "Major";
+                                    break;
+                                case "latestminor":
+                                    rollForward = "LatestMinor";
+                                    break;
+                                case "minor":
+                                    rollForward = "Minor";
+                                    break;
+                                case "latestpatch":
+                                    rollForward = "LatestPatch";
+                                    break;
+                                case "disable":
+                                    rollForward = "Disable";
+                                    break;
+                                default:
+                                    this.Messaging.Write(ErrorMessages.IllegalAttributeValueWithLegalList(sourceLineNumbers, element.Name.LocalName, attrib.Name.LocalName, rollForward, "latestmajor, major, latestminor, minor, latestpatch and disable"));
+                                    break;
+                            }
+                            break;
+                        default:
+                            this.ParseHelper.UnexpectedAttribute(element, attrib);
+                            break;
+                    }
+                }
+                else
+                {
+                    this.ParseHelper.ParseExtensionAttribute(this.Context.Extensions, intermediate, section, element, attrib);
+                }
+            }
+
+            if (null == id)
+            {
+                id = this.ParseHelper.CreateIdentifier("ndncc", runtimeType, platform, version, variable);
+            }
+
+            if (String.IsNullOrEmpty(runtimeType))
+            {
+                this.Messaging.Write(ErrorMessages.ExpectedAttribute(sourceLineNumbers, element.Name.LocalName, "RuntimeType"));
+            }
+
+            if (String.IsNullOrEmpty(platform))
+            {
+                this.Messaging.Write(ErrorMessages.ExpectedAttribute(sourceLineNumbers, element.Name.LocalName, "Platform"));
+            }
+
+            if (String.IsNullOrEmpty(version))
+            {
+                this.Messaging.Write(ErrorMessages.ExpectedAttribute(sourceLineNumbers, element.Name.LocalName, "Version"));
+            }
+
+            this.ParseHelper.ParseForExtensionElements(this.Context.Extensions, intermediate, section, element);
+
+            this.ParseHelper.CreateCustomActionReference(sourceLineNumbers, section, "Wix4NetFxDotNetCompatibilityCheck", this.Context.Platform, CustomActionPlatforms.ARM64 | CustomActionPlatforms.X64 | CustomActionPlatforms.X86);
+
+            if (!this.Messaging.EncounteredError)
+            {
+                // TODO: This will eventually be supported under Module/Product, but is not yet.
```

Comment:
I assume that you actually meant supported under Bundle.

@robmen @barnson We haven't really discussed how this would work. My initial guess would be for it to work kind of like the `SoftwareTag` element. When the parent element is a `Fragment`, then `DotNetCompatibilityCheck` should either have the `Property` attribute (for Module/Product) or the `Variable` attribute (for Bundle). Both can't be specified. If it has the `Property` attribute, then the Bundle specific attributes (`Condition` and `After`) should not be specified.

If we go that direction, then for this PR it shouldn't be parsing `Variable`, `Condition`, or `After` and should be parsing `Property` instead.

Reply:
I actually copied the comment from `UtilCompiler.cs line 232` and my point was that if I uncomment the following line `this.ParseHelper.CreateWixSearchSymbol(...)`, while using `netfx:DotNetCompatibilityCheck` from within a `Product` I get the following: `warning WIX1150: The binder doesn't know how to place the following symbol into the output: SymbolName: 'WixSearch'`. So since my initial comment in code was ambiguous I hope it's now clear what I meant.

Second thing @rseanhall based on your comment on the issue https://github.com/wixtoolset/issues/issues/6264#issuecomment-1265710694 I've focused on MSI. I only added `Condition` and `After` because the design doc specified them, and as you can see, I'm already using `Variable` as a `Property`. So, since I've fixed most of the other comments you wrote on PR, once you decide how this should work let me know so I can make required changes and push everything for the second PR review iteration.

Reply:
As far as I know, we don't currently have an element that can be authored in a fragment and that can either be used in an MSI or bundle which is why I included @robmen and @barnson. I was suggesting that the `Property` attribute would tell the compiler that it is for an MSI (don't create the search symbol) and that the `Variable` attribute would tell the compiler that it's for a bundle (do create the search symbol). The search symbol is really only for bundles. 

`Condition` and `After` are in the design doc because the initial plan was to have only one way to detect .NET Core. That meant that the element was going to be implemented for bundles, too. If the element doesn't actually use those attributes then it shouldn't be parsing them.

Reply:
I agree with @rseanhall here. Given that this is focused on MSI, use the `Property` attribute only and skip the Bundle related attributes.